byterun/domain.c

```diff
@@ -737,6 +738,80 @@ void caml_sample_gc_stats(struct gc_stats* buf)
   }
 }
 
+void caml_print_stats () {
+  struct gc_stats s;
+  int i;
+#ifdef COLLECT_STATS
+  struct detailed_stats ds;
+  caml_domain_state* st;
+  uint64 total;
+#endif
+
+  caml_gc_stat(Val_unit);
+  caml_sample_gc_stats(&s);
+  fprintf(stderr,"**** GC stats ****\n");
+  fprintf(stderr, "Minor words:\t\t%llu\n", s.minor_words);
```

Comment:
I wrote a comment about the portability of `%llu`, but the portable version is really ugly, and `%llu` is widespread enough these days that the current code is probably fine. If you care, here's the comment:

`%llu` is slightly dodgy for `uint64`. C99 specifies `ll` for `long long`, which is generally but not necessarily 64 bits. Historically, MSVC demands `I64` (e.g. `%I64u`) for 64-bit numbers, but I think it supports `ll` now as well.

The portable way to do this uses `ARCH_INT64_PRINTF_FORMAT` from config.h:

    fprintf(stderr, "Minor words:\t\t%"ARCH_INT64_PRINTF_FORMAT"u\n", s.minor_words);

(note the quotes and the `u`)